### ISSUE TITLE
Onyx bolt (e) effect being calculated incorrectly

### ISSUE THREAD

**Issue by jmyaeger** [weirdgloop/osrs-dps-calc#94]
### What went wrong?

The max hit of a crossbow + onyx bolts (e) is much higher than it should be. 

### What did you expect to happen?

It should be increasing the max hit by 20% (or 32% with ZCB).

### What browsers are you seeing the problem on?

Chrome

### What device(s) are you seeing the problem on?

Windows

### Any other information

Onyx bolts appear to implemented in the same way as dragonstone bolts in the code, adding flat extra damage based on visible ranged level:

```
if (this.wearing(['Onyx bolts (e)', 'Onyx dragon bolts (e)']) && !mattrs.includes(MonsterAttribute.UNDEAD)) {
        const chance = 0.1 * kandarinDiaryFactor;
        const effectDmg = Math.trunc(rangedLvl * (zaryte ? 32 : 20) / 100);
        dist = new AttackDistribution([
          new HitDistribution([
            ...standardHitDist.scaleProbability(1 - chance).hits,
            ...HitDistribution.linear(acc, effectDmg, max + effectDmg).scaleProbability(chance).hits,
          ]),
        ]);
      }
```
This should instead be similar to the diamond bolt implementation (but taking accuracy into account). 

I should also point out that the bolt proc chance here is `0.1 * kandarinDiaryFactor`, but per the wiki, it should be `0.11 * kandarinDiaryFactor` in PvM.

**Comment by LlemonDuck** [comment]
From the wiki it's a bit unclear:
> Enchanted onyx bolts have a chance of triggering the Life Leech effect. This will deal 20% extra damage (32% when under the effect of the Zaryte crossbow)

IIRC, the diamond bolts page specifically specifies that it raises the max hit. Do you have something backing this up? Then I can also include it in a comment in the code

**Comment by jmyaeger** [comment]
I've never tested it, but it should be fairly simple to test—if an onyx bolt procs and deals less than 20% of my visible ranged level in damage, it must be multiplying the max hit. I can try it out and report back.

The Bitterkoekje calc implements it as multiplying the max hit, FWIW. Not sure if that's verified anywhere or if that's just how they interpreted it as well.

**Comment by jmyaeger** [comment]
Well that was lucky, I only needed one hit to confirm this—my first hit on a monster proc'd an onyx bolt, and it hit an 8. At 112 visible ranged, the onyx proc would hit a minimum of 22 if it tacked on extra damage like dragonstone bolts.

**Comment by LlemonDuck** [comment]
Great, I'll take your word for it. Thanks for testing!

**Comment by jmyaeger** [comment]
Hey, was just reading through the code and saw that the onyx proc chance is still set at 10%—per the wiki, it should be 11% in PvM (but 10% in PvP).

**Comment by LlemonDuck** [comment]
Thank you, fixed in 9e82270f8e8ea7ef98d4a1093377aa42919c3ce3

**Comment by jmyaeger** [comment]
Resurrecting this because I just noticed that the onyx effect's max hit is currently implemented as 
`const effectMax = max + Math.trunc(rangedLvl * (zaryte ? 32 : 20) / 100);`

I haven't tested the max hit to confirm, but given the wording of dealing 20% or 32% extra damage (with no mention of the player's visible ranged level), I would be surprised if this shouldn't actually be
`const effectMax = Math.trunc(max * (zaryte ? 132 : 120) / 100);`
